baseapp/abci.go

```diff
@@ -379,9 +379,17 @@ func (app *BaseApp) snapshot(height int64) {
 
 // Query implements the ABCI interface. It delegates to CommitMultiStore if it
 // implements Queryable.
-func (app *BaseApp) Query(req abci.RequestQuery) abci.ResponseQuery {
+func (app *BaseApp) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 	defer telemetry.MeasureSince(time.Now(), "abci", "query")
 
+	// Add panic recovery for all queries.
```

Comment:
Sorry if this is a silly question but why do we need panic recovery in client-facing code? Where in the executional flow path of client-facing code do we panic and why?

Reply:
See https://github.com/cosmos/cosmos-sdk/issues/8028. One of the gRPC queries panics.

I agree with you that we need more testing, but even then there might be some unexpected panics (like that nil pointer). Hence adding a recovery just in case.

Reply:
I see what you mean -- this is for non-explicit `panic`s. Either way, it can't hurt to have this, but let's be sure to not have any _explicit_ `panic`s in client-facing code 👍